File: 0/1/0/010.c
// Copyright (C) 2013 Jorge Aparicio

#include <stdbool.h>
#include <stdio.h>
#include <stdlib.h>

static int size = 1000000;

int main() {
  long i, j, acc = 2, p;
  bool *sieve = (bool *)calloc(size, sizeof(char));

  for (i = 0; i < size; i++) {
    if (!sieve[i]) {
      p = 2*i + 3;

      acc += p;

      for (j = p*p; j < 2*size + 3; j += 2*p)
        sieve[(j - 3) / 2] = true;
    }
  }

  free(sieve);

  printf("%llu\n", acc);

  return 0;
}

File: 0/1/0/010.rs
// Copyright (C) 2014 Jorge Aparicio

use std::iter::range_step;

static size: int = 1000000;

fn main() {
  let mut sieve: [bool, ..size] = [true, ..size];
  let mut acc = 2;

  for i in range(0, size) {
    if sieve[i] {
      let p = 2*i + 3;

      acc += p;

      for j in range_step(p * p, 2*size + 3, 2*p) {
        sieve[(j - 3) / 2] = false;
      }
    }
  }

  println!("{}", acc);
}

